lib/sdk/auth.js

```
@@ -0,0 +1,124 @@
+			oauth_consumer_key: request.sdk.transport.config.apiKey,
+			oauth_token: oauthToken,
+			oauth_verifier: oauthVerifier
+		},
+		decoratedRequest = request,
+		definition = request.sdk.findDefinition(apiV1, 'getAccessToken');
+
+	decoratedRequest.method = definition.verb;
+	decoratedRequest.url = generateURL(definition);
+
+	oauthParams['oauth_signature'] = request.sdk.transport.createSignature(decoratedRequest, oauthParams, oauthTokenSecret);
+
+	return request.sdk.transport.call(
+		request.sdk.findDefinition(apiV1, 'getAccessToken'),
+		oauthParams,
+		{} // no additional params
+	)
+	.then(function (response) {
+		// clean up the response for consumers
+		return response.body;
+	});
+}
+
+module.exports = function () {
+
+	var request = this;
+
+	if (process.browser) {
+		throw new Error('Authentication is not supported in the browser at this time.');
+	}
+
+	return {
+
+		prepareRequestToken: function (oauthCallbackURI) {
+			if (process.browser) {
+				console.warn('It is stringly discouraged to make this call from a browser. See github.com/flickr/flickr for more details.');
+			}
+
+			return getRequestToken(request, oauthCallbackURI);
+		},
+
+		/**
+		 * We handle the auth pop up ourself, or defer to the user
+		 * The callback page in the popup will beacon back the oauth token
+		 * We then use that to request an access token
+		 * And when that comes back we're authed!
+		 */
+		authenticateUser: function (requestToken, requestTokenSecret, requestTokenVerifier) {
+
+			return new Promise(function (resolve, reject) {
+
+				var requestTokenDefinition = request.sdk.findDefinition(apiV1, 'getRequestToken');
+				var requestURL = requestTokenDefinition.schema.schemes[0] + '://' + requestTokenDefinition.schema.host + requestTokenDefinition.schema.basePath + '/oauth/authorize?oauth_token=' + requestToken;
+
+				if (!process.browser) {
+					if (!requestTokenVerifier) {
+						return new Error('You must specify the request token verifier.');
```

`reject` instead of `return`?